pkg/runctl/run_manager.go

```diff
@@ -97,60 +98,88 @@ func (c *runManager) prepareRunNamespace(pipelineRun k8s.PipelineRun) error {
 	accountManager := k8s.NewServiceAccountManager(c.factory, runNamespace)
 
-	serviceAccount, err := accountManager.CreateServiceAccount(serviceAccountName, scmCloneSecretName, pullSecretName)
+	serviceAccount, err := accountManager.CreateServiceAccount(serviceAccountName, pipelineCloneSecretName, imagePullSecrets)
 	if err != nil {
-		return errors.Wrap(err, "Failed to create service account.")
+		return errors.Wrap(err, "failed to create service account.")
 	}
 
 	//Add Role Binding to Service Account
 	_, err = serviceAccount.AddRoleBinding(runClusterRoleName, runNamespace)
 	if err != nil {
-		return errors.Wrap(err, "Failed to create role binding")
+		return errors.Wrap(err, "failed to create role binding")
 	}
 
 	return nil
 }
 
-func (c *runManager) copySecrets(targetNamespace string, secretNames []string, pipelineRun k8s.PipelineRun) error {
-	for _, secretName := range secretNames {
-		targetClient := c.factory.CoreV1().Secrets(targetNamespace)
-		secret, err := c.secretProvider.GetSecret(secretName)
-		if err != nil {
-			pipelineRun.UpdateResult(v1alpha1.ResultErrorContent)
-			pipelineRun.UpdateMessage(err.Error())
-			return err
-		}
-		createSecret(targetClient, targetNamespace, secretName, secret)
+func (c *runManager) copyPipelinePullSecret(pipelineRun k8s.PipelineRun, secretHelper secretUtils.SecretHelper) (string, error) {
+	pipelineCloneSecret := pipelineRun.GetSpec().JenkinsFile.Secret
+	if pipelineCloneSecret == "" {
+		return "", nil
 	}
-	return nil
+	random, err := utils.RandomAlphaNumString(6)
+	if err != nil {
+		return "", err
+	}
+	repoServer, err := pipelineRun.GetRepoServerURL()
+	if err != nil {
+		return "", err
+	}
+	transformers := []secrets.SecretTransformerType{
+		secrets.StripAnnotationsFunc("jenkins"),
+		secrets.StripLabelsFunc("jenkins"),
+		secrets.AppendNameSuffixFunc(random),
+		secrets.SetAnnotationFunc("tekton.dev/git-0", repoServer),
+	}
+	names, err := c.copySecrets(secretHelper, []string{pipelineCloneSecret}, pipelineRun, nil, transformers...)
+	if err != nil {
+		return "", err
+	}
+	return names[0], nil
 }
 
-func createSecret(client corev1.SecretInterface, namespace string, name string, secret *v1.Secret) {
-	newSecret := &v1.Secret{Data: secret.Data, StringData: secret.StringData, Type: secret.Type}
-	newSecret.SetName(name)
-	newSecret.SetNamespace(namespace)
-	newSecret.SetLabels(secret.GetLabels())
-	newSecret.SetAnnotations(secret.GetAnnotations())
-	_, err := client.Create(newSecret)
+func (c *runManager) copySecrets(secretHelper secretUtils.SecretHelper, secretNames []string, pipelineRun k8s.PipelineRun, filter secrets.SecretFilterType, transformers ...secrets.SecretTransformerType) ([]string, error) {
+	storedSecretNames, err := secretHelper.CopySecrets(secretNames, filter, transformers...)
 	if err != nil {
-		log.Printf("Cannot create secret '%s' in namespace '%s': %s", name, namespace, err)
+		pipelineRun.UpdateResult(v1alpha1.ResultErrorContent)
```

Comment:
Why `ResultErrorContent`? It could be an infrastructure error, too. How do we distinguish them?

Reply:
See 462b22b5d151345580f36946ad7690b09b6bf6d2